diff -ru swift-orig/lib/Basic/Version.cpp swift/lib/Basic/Version.cpp
--- swift-orig/lib/Basic/Version.cpp	2017-08-07 12:21:34.150036560 +1000
+++ swift/lib/Basic/Version.cpp	2017-08-07 12:22:07.662904723 +1000
@@ -386,31 +386,7 @@
   OS << SWIFT_VENDOR " ";
 #endif
 
-  OS << "Swift version " SWIFT_VERSION_STRING;
-#ifndef SWIFT_COMPILER_VERSION
-  OS << "-dev";
-#endif
-
-  if (!(effectiveVersion == Version::getCurrentLanguageVersion())) {
-    OS << " effective-" << effectiveVersion;
-  }
-
-#if defined(SWIFT_COMPILER_VERSION)
-  OS << " (swiftlang-" SWIFT_COMPILER_VERSION;
-#if defined(CLANG_COMPILER_VERSION)
-  OS << " clang-" CLANG_COMPILER_VERSION;
-#endif
-  OS << ")";
-#elif defined(LLVM_REVISION) || defined(CLANG_REVISION) || \
-      defined(SWIFT_REVISION)
-  OS << " (";
-  printFullRevisionString(OS);
-  OS << ")";
-#endif
-
-  // Suppress unused function warning
-  (void)&printFullRevisionString;
-
+  OS << "Swift version " SWIFT_VERSION_STRING " (swift-4.0-DEVELOPMENT)";
   return OS.str();
 }
 
diff -ru swift-orig/utils/build-script-impl swift/utils/build-script-impl
--- swift-orig/utils/build-script-impl	2017-08-07 12:21:34.345035793 +1000
+++ swift/utils/build-script-impl	2017-08-07 12:24:16.510397847 +1000
@@ -2431,10 +2431,13 @@
                 fi
     
                 with_pushd "${FOUNDATION_SOURCE_DIR}" \
+                    export CFLAGS=-I${FOUNDATION_SOURCE_DIR}/closure
+                with_pushd "${FOUNDATION_SOURCE_DIR}" \
                     call env SWIFTC="${SWIFTC_BIN}" CLANG="${LLVM_BIN}"/clang SWIFT="${SWIFT_BIN}" \
                       SDKROOT="${SWIFT_BUILD_PATH}" BUILD_DIR="${build_dir}" DSTROOT="$(get_host_install_destdir ${host})" PREFIX="$(get_host_install_prefix ${host})" ./configure "${FOUNDATION_BUILD_TYPE}" ${FOUNDATION_BUILD_ARGS[@]} -DXCTEST_BUILD_DIR=${XCTEST_BUILD_DIR} $LIBDISPATCH_BUILD_ARGS
                 with_pushd "${FOUNDATION_SOURCE_DIR}" \
                     call ${NINJA_BIN}
+                    export CFLAGS=
 
                 # Foundation builds itself and doesn't use cmake
                 continue
Only in swift/utils: build-script-impl.rej
